src/main/scala/ie/zalando/fabric/gateway/service/IngressDerivationChain.scala

```diff
@@ -90,8 +90,33 @@ class IngressDerivationChain(stackSetOperations: StackSetOperations)(implicit ma
       val finalRoutes = gateway.corsConfig.fold(skipperRoutes) { corsConfig =>
         withCors(gateway, meta.name, corsConfig, skipperRoutes)
       }
-      combineBackendsAndRoutes(backends, finalRoutes, meta)
+      val ingressDefinitions = combineBackendsAndRoutes(backends, finalRoutes, meta)
+
+      gateway.serviceProvider match {
+        case StackSetProvidedServices(_, _) if versionedHostsEnabled => appendServiceSpecificHosts(ingressDefinitions)
+        case _ => ingressDefinitions
+      }
+    }
+  }
+
+  def appendServiceSpecificHosts(ingresses: List[IngressDefinition]): List[IngressDefinition] = {
+    val serviceSpecificIngresses = for {
+      ingress <- ingresses
+      hostMapping <- ingress.hostMappings
+      splitHost = hostMapping.host.split('.')
+      if splitHost.length == 4 && splitHost(2) == "zalan" && splitHost(3) == "do"
```

Comment:
Can you explain what's happening here? Seems like we're binding ourselves to a host pattern... 🤔

Reply:
Yeah I said the same in a comment a few minutes ago upon looking at it again. Will change it

Reply:
Ok sorry, just saw that now. Maybe have the subdomains passed in via env var. Then it can be whatever people want

Reply:
`Maybe have the subdomains passed in via env var` do you mean like the above `svc.$baseUrl` or something different?